sdk/storage/azure-storage-file-share/swagger/src/main/java/ShareStorageCustomization.java

```diff
@@ -274,4 +277,54 @@ private static void addMissingHashMapImport(PackageCustomization implementationM
             implementationModels.getClass(className).addImports("java.util.HashMap");
         }
     }
+
+    //Fix datetime conversion in FilesCreateHardLinkHeaders
```

Comment:
when I ran my hard link test, I encountered a conversion related error [here](https://github.com/Azure/azure-sdk-for-java/blob/047add37cc44813a3c7d30030ba3b36499e7ad72/sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/models/FilesCreateHardLinkHeaders.java#L133). It comes down to the DateTimeRfc1123 constructor requiring the string to be in RFC1123 (to convert it into OffsetDateTime, more info [here](https://github.com/Azure/azure-sdk-for-java/blob/709d1c9da8aa94e8965c78d90e5d55074aadd647/sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java#L81))- but the value we receive from the `xMsFileCreationTime` header is already in OffsetDateTime format, so it throws an error. We will also have to do the same swagger transform when we implement the symbolic link feature, as it suffers the same issue. I'm not sure if this generation is intended, as most of the other header classes besides those added for this feature use `new OffsetDateTime(xMsFileCreationTime)` and not `new DateTimeRfc1123(xMsFileCreationTime)`, but I talked to Sean and he said to go ahead with the swagger transform. 

In terms of moving forwards with the customization, I can probably find a way to find+replace just the problematic if blocks instead of the whole method